proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md

```diff
@@ -226,39 +204,43 @@ The above code is sugar for:
 
 ```swift
 func insert(text: String) {
-  @Logged var text = text
+  let _text: Logged<String> = Logged(wrappedValue: text)
+
+  var text: String { _text.wrappedValue }
 }
 ```
 
-which the compiler further desugars to generate:
+Note that the backing storage is a let-constant, and the local `text` property does not have a setter.
 
-```swift
-func insert(text: String) {
-  var _text: Logged<String> = Logged(wrappedValue: text)
+> **Rationale**: It would be confusing if it appears that a user can mutate a parameter, because those changes will not be reflected in the caller once the function returns.
 
-  var text: String { _text.wrappedValue }
-}
-```
+Implementation-detail property wrappers on parameters must support initialization from a wrapped value, and the parameter type must be equal to the wrapped value type of the wrapper.
+
+##### Arguments in the property-wrapper attribute
+
+Property-wrapper attributes with arguments applied to parameters are always implementation-detail property wrappers, even if the property wrapper supports initialization from a projected value.
```

Comment:
> [...] implementation-detail property wrappers

Should we perhaps shift the focus from the wrapper to its initialization? The current semantics render API significance dependent on the property-wrapper initializer to be called. Thus, the above could be written as:

> Property-wrapper attributes with arguments applied to parameters always ~are~ **result in** implementation-detail ~property wrappers~ **initialization** , even if the property wrapper supports initialization from a projected value.

Reply:
There is no difference between an "implementation-detail property wrapper" and "implementation-detail initialization". I don't think changing the wording here makes anything clearer